src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs

```diff
@@ -2430,5 +2431,28 @@ private bool ConsumeMultiLineComment(ReadOnlySpan<byte> localBuffer, int previou
             _tokenType = JsonTokenType.Comment;
             return true;
         }
+
+        private string DebuggerDisplay => $"TokenType = {DebugTokenType} (TokenStartIndex = {TokenStartIndex}) Consumed = {BytesConsumed}";
+
+        // Using TokenType.ToString() (or {TokenType}) fails to render in the debug window. The
+        // message "The runtime refused to evaluate the expression at this time." is shown. This
+        // is a workaround until we root cause and fix the issue.
+        private string DebugTokenType
```

Comment:
This seems odd. Would `nameof()` work?

Reply:
It does, changing it.

Reply:
I am still curious to know why other approaches don't work. cc @jkotas, @cshung - any ideas?

Reply:
My guess is that it is because of `Utf8JsonReader` is a ref struct. The debugger has limitations on what it can do with ref structs. Maybe worth sending issue on this to VS debugger team to make sure that it is the know limitation.

Reply:
I can't get a smaller repro with a ref struct which makes me lean towards "bug" here. Unfortunately I haven't been able to figure out the magic sauce, so I plan to follow up with a bug once this is in.

Reply:
Adding @gregg-miskelly